### PR TITLE
prepare backend for adding fooditems

### PR DESCRIPTION
**Pull request by ingbeck**

### CODE REVIEW

File: backend/src/main/java/com/github/ingbeck/backend/service/DiaryService.java [hunk 1]
```diff
@@ -6,17 +6,54 @@
...
+            DiaryEntry updatedDiaryEntry = diaryEntryToUpdate.withFoodItems(updatedFoodItems).withTotalCalories(totalCalories);
+            newDiaryEntries = currentDiaryEntries.stream().map(entry -> entry.date().equals(date) ? updatedDiaryEntry : entry).toList();
+
+        }catch (NoSuchElementException e){
+            totalCalories = newFoodItems.stream().map(FoodItem::calories).reduce(0, Integer::sum);
+            newDiaryEntries = List.of(new DiaryEntry(date, newFoodItems, totalCalories));
+
+        }
+
+        return diaryRepository.save(getDiaryByUserId(userId).withDiaryEntries(newDiaryEntries));
```
Comment (MaHambach) [inline backend/src/main/java/com/github/ingbeck/backend/service/DiaryService.java:56]: You return the whole diary, when an entry is updated?

File: backend/src/test/java/com/github/ingbeck/backend/controller/DiaryControllerTest.java [hunk 2]
```diff
@@ -34,4 +45,104 @@ void createNewDiary_whenCalledWithUserId_thenReturnNewDiaryWithUserId() throws E
...
+                                "amount": 1,
+                                "unit": "",
+                                "calories": 54,
+                                "mealType": "DINNER"
+                              }
+                         ]
+                         """))
+                .andExpect(status().isOk())
+                .andExpect(content()
+                        .json("""
```
Comment (MaHambach) [inline backend/src/test/java/com/github/ingbeck/backend/controller/DiaryControllerTest.java:69]: I recommend objectMapper. In my opinion it improves the readability a lot.
Example from my code:
```
AppUserRegister userRegister = new AppUserRegister("user", "password");
MvcResult userJson = mvc.perform(post("/api/users/register")
                        .contentType(MediaType.APPLICATION_JSON)
                        .content(objectMapper.writeValueAsString(userRegister)))
                .andExpect(status().isCreated())
                .andReturn();
AppUser user = objectMapper.readValue(userJson.getResponse().getContentAsString(), AppUser.class)
```
In this example a user gets created and returned. Afterwards I got the returned user and can check all the things I might want to check.

**Review by MaHambach — APPROVED** [review]
LGTM. Two small remarks/ questions.

File: backend/src/test/java/com/github/ingbeck/backend/controller/DiaryControllerTest.java (hunk 2 above)
Comment (ingbeck) [reply backend/src/test/java/com/github/ingbeck/backend/controller/DiaryControllerTest.java:69]: THX. I will consider to refactor my tests.

File: backend/src/main/java/com/github/ingbeck/backend/service/DiaryService.java (hunk 1 above)
Comment (ingbeck) [reply backend/src/main/java/com/github/ingbeck/backend/service/DiaryService.java:56]: I need the whole diary up to date, so the user can switch between the diary entries.

**Pull request merged**